controller.py

```diff
@@ -18,9 +19,34 @@
 initial_temp = 2.
 cool = 0.9
 reanneal = 100
-iterr = 100000
 nswaps = 3
 
-solution, history = simulated_annealing(graph, distance, initial_path, initial_temp, cool, reanneal, iterr, changepath, nswaps)
-print "Calculated path: " + str(solution)
-print "Calculated path length: " + str(distance(graph, solution))
+# Run simulated annealing
+print "\nSimulated Annealing\n"
+for iterr in [10**x for x in [3,4,5]]:
+	with Timer() as t:
+		solution, history = simulated_annealing(graph, distance, initial_path, 
+												initial_temp, cool, reanneal, iterr, 
+												changepath, nswaps)
+	print "\nIterations: {:.2E}".format(iterr)
+	print "Calculated path: " + str(solution)
+	print "Calculated path length: " + str(distance(graph, solution))
+	print "Time: " + str(t.interval)
+
+# Initial values for parallel tempering
+nsystems = 3
+initial_paths = [np.random.permutation(size) for i in xrange(nsystems)]
+initial_temps = [1., 5., 10.]
+nbefore = 100
+
+# Run parallel tempering
+print "\nParallel Tempering\n"
+for iterr in [10**x for x in [3,4,5]]:
+	with Timer() as t:
+		solution, history = parallel_tempering(graph, distance, initial_paths, initial_temps,
+											   iterr, changepath, nswaps, nbefore)
+
+	print "\nIterations: {:.2E}".format(iterr)
+	print "Calculated path: " + str(solution)
+	print "Calculated path length: " + str(distance(graph, solution)) 
+	print "Time: " + str(t.interval)
```

Comment:
With 10^5 iterations and exchanging systems every 100 iterations, simulated annealing takes 13 seconds and parallel tempering takes 38 seconds.